### DISCUSSION TITLE
Stb-nmc-400 error messages (KPN ITV / UDM Pro)

### DISCUSSION THREAD

**Discussion by jprinsen** [fabianishere/udm-iptv#325] [Q&A]
Hi all, I have an Unifi UDM Pro with KPN itv on it running on VLAN 4 (all hard-wired). Occassionally stb-nmc-400 error messages pop up on my decoder, usually without any reason and also when we are just watching (and not changing channels). Only possibility is restarting the decoder(s) which gives us approximately 2 days of screen viewing without error messages.  

I have searched on the net, did multiple re-installations and changed settings in Unifi UDM (such as de-activation of IG Proxy, change of VLANs, opening up firewall settings back and forth to VLAN 4 etc etc). All without any luck. The error messages keep popping up after .. ~2 days.

I would be extremely grateful to the person that has the golden tip or solution so that these error message could belong to the past.

Diagnose-log:
```
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br0  br4
IGMP Proxy quickleave disabled: true
IGMP Proxy debug: true
=== IP Link and Route ===
53: iptv@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.181.120.180/22 brd 10.181.123.255 scope global iptv
       valid_lft forever preferred_lft forever
10.181.120.0/22 proto kernel scope link src 10.181.120.180 
213.75.112.0/21 via 10.181.120.1 metric 253 
=== Service Logs ===
May 29 20:38:45 HSL28 udm-iptvd[2982725]: DEBUG[imp_input_report_v1v2@289]: group address 239.255.255.250
May 29 20:38:45 HSL28 udm-iptvd[2982725]: INFO[imp_verify_multicast_addr@262]: Group address 239.255.255.250 belongs to IPv4 Local Scope.
May 29 20:38:45 HSL28 udm-iptvd[2982725]: DEBUG[main@653]: Data arrive
May 29 20:38:45 HSL28 udm-iptvd[2982725]: DEBUG[mcast_recv_igmp@748]: src addr = 192.168.1.254
May 29 20:38:45 HSL28 udm-iptvd[2982725]: received interface  15
May 29 20:38:45 HSL28 udm-iptvd[2982725]: DEBUG[imp_input_report_v3@353]: type : 0x22
May 29 20:38:45 HSL28 udm-iptvd[2982725]:  num p_grec 2
May 29 20:38:45 HSL28 udm-iptvd[2982725]: DEBUG[imp_input_report_v3@370]: group is = 239.255.255.250
May 29 20:38:45 HSL28 udm-iptvd[2982725]: INFO[imp_verify_multicast_addr@262]: Group address 239.255.255.250 belongs to IPv4 Local Scope.
May 29 20:38:45 HSL28 udm-iptvd[2982725]: DEBUG[imp_input_report_v3@370]: group is = 224.0.0.251
```

Unifi UDM Pro specifications:
> UniFi OS Version 3.2.12
> Unifi Network 8.1.127
> IPTV on VLAN 4 (Default Network VLAN ID 1
> Multicast DNS on Default and IPTV
> IGMP Snooping on Default and IPTV
> IGMP Proxy de-activated/off
> DHCP Snooping turned ON.

Multiple KPN 4K-decoders (VIP 5202); all with latest software on it and hard-wired. 

Many thanks in anticipation!

**Comment by Bolten88** [comment]
Has it worked without problems? It is recommended to have the decoders on a separate vlan. I would try that as a first step.

**Reply by jprinsen** [reply]
Thanks Bolten88. The decoders are all on their own - dedicated - VLAN (VLAN 4). I have the feeling that the error codes pop up more frequent lately.

**Reply by Bolten88** [reply]
I misunderstood that. Is is possible that there is a conflict with vlan 4 on the wan side? I have the the KPN Tv box on vlan 20, and i don’t use vlan 4 at all on the LAN side. You could try using another vlan id for the tv box lan. Re run the install script after that to configure the new vlan id. You could at least try it.

**Reply by jprinsen** [reply]
You're right @Bolten88 - it is certainly worth a try. Did change VLAN of the decoders to VLAN 30 and subsequently reconfigured the script. I will keep you all posted about the developments (might take a couple of days though due to resetting the decoders).

**Reply by jprinsen** [reply]
Okay, In three days time I experienced three stb-nmc-400 error messages; substantially less than before but still not completely error free; strange thing is that after, say, 1 minute of freeze the stb-nmc-400 seems to disappear. Since there is somewhat longer channel freezing upon swithing tv channels, I wonder whether this may (also?) be related to the ARP Timeout settings of the UDM Pro (currently pushed to 600 ms; custom setting at the UDM device tab in Unifi).
